crates/python/src/py_sync.rs

```diff
@@ -20,22 +20,26 @@
 /// ```
 macro_rules! py_sync {
     ($py: ident, $body: expr) => {{
-        let runtime = ::pyo3_asyncio::tokio::get_runtime();
-        let handle = runtime.spawn($body);
+        $py.allow_threads(|| {
```

Comment:
Per the [`pyo3_log` documentation](https://docs.rs/pyo3-log/latest/pyo3_log/#interaction-with-python-gil), we need to wrap this function with `allow_threads` to avoid deadlocks. This necessitates using `Python::with_gil` to check the signals. In my testing so far, that's worked as expected.